### ISSUE TITLE
Ignore file even when passed as CLI argument

### ISSUE THREAD

**Issue by dmateusp** [sqlfluff/sqlfluff#483]
Currently sqlfluff ignores files that match patterns in `.sqlfluffignore` but only if a file is not specified by name directly. 

**For example**

.sqlfluffignore:
```
test.sql
```

test.sql:
```
select a, b, c, d, e fromm table_x
```

Result of `sqlfluff lint .` is empty while result of `sqlfluff lint test.sql` is :
```
== [test.sql] FAIL
L:   1 | P:  22 | L012 | Implicit aliasing of column not allowed. Use explicit
                       | `AS` clause.
L:   1 | P:  27 |  PRS | Found unparsable section: ' table_x'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
```

### Why is it a problem?

I'm currently using sqlfluff with `pre-commit`, which passes a list of changed files to the processes so only the changed files are linted/validated etc. Right now I need to separately add an `exclude:` to the pre-commit config as such:

```yaml
exclude: test.sql
```

### Suggested solution

By default I think sqlfluff should always ignore files that match the patterns defined in `.sqlfluffignore`. If a file is passed explicitly I think it should just log a warning that the file was not linted/parsed because it was ignored. A `--force` flag could be passed to ignore the `.sqlfluffignore` file.

**Comment by alanmcruickshank** [comment]
Sounds like a good idea. Great clear issue description, and a pragmatic solution.

I like the idea of a warning if the file is specified exactly but still skipped.

The only bit I'm not sure about is exactly what the name of the `--force` flag should be. There are a couple of things that could be _forced_ at the moment, and so I think we should be more specific in what the file name is that is being forced.

@dmateusp - want to have a go at this one?

**Comment by dmateusp** [comment]
you're right, `--force` is probably not great naming! maybe a `--not-ignore-file` argument that could be repeated?

I'll try to spend some time on that issue
